### PR TITLE
[skipci]Merge block storage and file storage compilation scripts

### PR DESCRIPTION
**Pull request by linshiyx**
### What problem does this PR solve?

Issue Number: #2026 

Problem Summary:

### What is changed and how it works?

What's Changed:
Compile curvebs: cd curve && make build stor=bs dep=1
Compile curvefs: cd curve && make build stor=fs dep=1

How it Works:
Combine compilation scripts of curvebs with curvefs

Side effects(Breaking backward compatibility? Performance regression?):
None

### Check List

- [x] Relevant documentation/comments is changed or added
- [x] I acknowledge that all my contributions will be made under the project's license

### CODE REVIEW

**Comment by ilixiaocui** [comment]
Please synthesize a commit for submission.

File: util/install.sh [hunk 1]
```diff
@@ -173,7 +199,41 @@ install_curvebs() {
...
+        local project_bin_filename=${BASH_REMATCH[3]}  # ex: curvefs_metaserver
+        local bazel_dir="$g_curve_dir/bazel-bin"
+        local binary="$bazel_dir/${BASH_REMATCH[1]}/$project_bin_filename"
+
+        # The below actions:
+        #   1) create project directory
+        #   2) copy binary to project directory
+        #   3) generate servicectl script into project directory.
+        create_project_dir $project_prefix
+        copy_file "$binary" "$project_prefix/sbin"
```
Comment (ilixiaocui) [inline util/install.sh:230]: If copying the file fails, for example the file does not exist, it is recomend to print an error and exit directly without continuing to execute.

File: util/install.sh [hunk 2]
```diff
@@ -226,6 +286,11 @@ install_etcd() {
     create_project_dir $project_prefix
     copy_file "$dst/etcd" "$project_prefix/sbin"
     copy_file "$dst/etcdctl" "$project_prefix/sbin"
+    gen_servicectl \
```
Comment (ilixiaocui) [inline util/install.sh:289]: ditto. Each `install_xx` judges whether the execution is successful, and terminates the script if it fails.

**Comment by ilixiaocui** [comment]
Thank for you contribute @linshiyx . I played around with the new build script and it's pretty good overall. And some small improvements can make this feature become better :)

File: util/build.sh [hunk 3]
```diff
@@ -174,6 +222,11 @@ build_requirements() {
 
 main() {
     get_options "$@"
+    get_version
+
+    if [[ "$g_stor" != "bs" && "$g_stor" != "fs" ]]; then
+        die "--stor must be either bs or fs\n"
```
[the remark this answers is not in the document]
Comment (linshiyx) [reply util/build.sh:228]: This error message instructs how to use build.sh, as the usage message does. Maybe I can update the script to print usage message when stor is not properly specified.

File: util/install.sh (hunk 1 above)
Comment (linshiyx) [reply util/install.sh:230]: This feature is already included in `create_project_dir`.

File: util/install.sh (hunk 2 above)
Comment (linshiyx) [reply util/install.sh:289]: This feature is already included in `gen_servicectl`.

File: util/build.sh (hunk 3 above)
[the remark this answers is not in the document]
Comment (ilixiaocui) [reply util/build.sh:228]: >  Maybe I can update the script to print usage message when stor is not properly specified.

good idea.

File: util/image.sh [hunk 4]
```diff
@@ -1,6 +1,7 @@
 #!/usr/bin/env bash
 
 # tmpl.sh = /usr/local/metaserver.conf /tmp/metaserver.conf
+# $1: bs/fs $2: tag $3: os
 function tmpl() {
```
Comment (ilixiaocui) [inline util/image.sh:5]: add  `set -e` to terminate any command execution error.

Comment (linshiyx) [reply util/image.sh:21]: Yeah that's an easier way. But I missed it and have done it referring to utils/build.sh :sweat_smile:

Comment (ilixiaocui) [reply util/image.sh:21]: It's ok :)

**Comment by ilixiaocui** [comment]
LGTM

**Review by ilixiaocui — APPROVED** [review]

**Review by cw123 — APPROVED** [review]

**Pull request merged**